core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -324,11 +324,20 @@ object LogConfig {
   }
 
   /**
+    * Check that the property values are valid relative to each other
+    */
+  def validateValues(props: Properties) {
+    require(props.getProperty(SegmentBytesProp).toLong <= props.getProperty(RetentionBytesProp).toLong,
```

Comment:
Need to check for `null` here or include the default value in this check. Some tests are failing because it's trying to convert a `null` to `Long`